R/test-that.r

```diff
@@ -73,6 +82,28 @@ test_code <- function(description, code, env) {
   invisible(ok)
 }
 
+report_results <- function(results) {
+  withRestarts(
+    raise_condition_from_result(results),
+    continue_test = function(e) NULL
+  )
+
+  invisible(results)
+}
+
+raise_condition_from_result <- function(results) {
+  if (results$passed) {
+    cond <- structure(list(results = results),
```

Comment:
Could we combine this object representation with the expectation, so there's only one type of thing to worry about?